### ISSUE TITLE
Allow modifying optimizer parameters

### ISSUE THREAD

**Issue by tadejsv** [jina-ai/finetuner#111]
At the very least, learning rate should be customizable.

We should also consider supporting multiple optimizers, e.g. Adam (which should be the default, as it is the most commonly used one), AdamW (used for transformers) and SGD, and setting their parameters.

Best way to do this is not clear - either we have some selection of optimizers that users can choose from, and we implement it in all frameworks, or we allow passing the optimizer object.
